common/spdat.cpp

```diff
@@ -2430,3 +2754,778 @@ bool AegolismStackingIsArmorClassSpell(uint16 spell_id) {
+		case BotSpellTypes::CompleteHeal:
+		case BotSpellTypes::PetCompleteHeals:
+		case BotSpellTypes::GroupCompleteHeals:
+			return false;
+		default:
+			return true;
+	}
+
+	return true;
+}
+
+bool IsResistanceOnlySpell(uint16 spell_id) {
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
+	const auto& spell = spells[spell_id];
+
+	for (int i = 0; i < EFFECT_COUNT; i++) {
+		if (IsBlankSpellEffect(spell_id, i)) {
+			continue;
+		}
+
+		if (
+			spell.effect_id[i] == SE_ResistFire ||
+			spell.effect_id[i] == SE_ResistCold ||
+			spell.effect_id[i] == SE_ResistPoison ||
+			spell.effect_id[i] == SE_ResistDisease ||
+			spell.effect_id[i] == SE_ResistMagic ||
+			spell.effect_id[i] == SE_ResistCorruption ||
+			spell.effect_id[i] == SE_ResistAll
+		) {
+			continue;
+		}
+
+		return false;
+	}
+
+	return true;
+}
+
+bool IsDamageShieldOnlySpell(uint16 spell_id) {
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
+	const auto& spell = spells[spell_id];
+
+	for (int i = 0; i < EFFECT_COUNT; i++) {
+		if (IsBlankSpellEffect(spell_id, i)) {
+			continue;
+		}
+
+		if (
+			spell.effect_id[i] != SE_DamageShield
+		) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+bool IsHateSpell(uint16 spell_id) {
+	if (
+		(IsEffectInSpell(spell_id, SE_Hate) && spells[spell_id].base_value[GetSpellEffectIndex(spell_id, SE_Hate)] > 0) ||
+		(IsEffectInSpell(spell_id, SE_InstantHate) && spells[spell_id].base_value[GetSpellEffectIndex(spell_id, SE_InstantHate)] > 0)
+	) {
+		return true;
+	}
+
+	return false;
+}
+
+bool IsCommandedSpellType(uint16 spell_type) {
+	switch (spell_type) {
+		case BotSpellTypes::Charm:
+		case BotSpellTypes::AEFear:
+		case BotSpellTypes::Fear:
+		case BotSpellTypes::Resurrect:
+		case BotSpellTypes::AELull:
+		case BotSpellTypes::Lull:
+		case BotSpellTypes::Teleport:
+		case BotSpellTypes::Succor:
+		case BotSpellTypes::BindAffinity:
+		case BotSpellTypes::Identify:
+		case BotSpellTypes::Levitate:
+		case BotSpellTypes::Rune:
+		case BotSpellTypes::WaterBreathing:
+		case BotSpellTypes::Size:
+		case BotSpellTypes::Invisibility:
+		case BotSpellTypes::MovementSpeed:
+		case BotSpellTypes::SendHome:
+		case BotSpellTypes::SummonCorpse:
+		//case BotSpellTypes::Cure:
+		//case BotSpellTypes::GroupCures:
+		//case BotSpellTypes::DamageShields:
+		//case BotSpellTypes::PetDamageShields:
+		//case BotSpellTypes::ResistBuffs:
+		//case BotSpellTypes::PetResistBuffs:
+			return true;
+		default:
+			return false;
+	}
+
+	return false;
+}
+
+bool IsPullingSpellType(uint16 spell_type) {
+	switch (spell_type) {
+		case BotSpellTypes::Nuke:
+		case BotSpellTypes::Lifetap:
+		case BotSpellTypes::Snare:
+		case BotSpellTypes::DOT:
+		case BotSpellTypes::Dispel:
+		case BotSpellTypes::Slow:
+		case BotSpellTypes::Debuff:
+		case BotSpellTypes::Stun:
+		case BotSpellTypes::HateLine:
+			return true;
+		default:
+			return false;
+		}
+
+	return false;
+}
+
+uint16 GetCorrectSpellType(uint16 spell_type, uint16 spell_id) {
```

Comment:
Commented in a few places here but there's quite a few opportunities to drop `else` and branching and returning early